lib/python/bind_slice_methods.h

```diff
@@ -270,31 +270,10 @@ void bind_slice_methods(pybind11::class_<T, Ignored...> &c) {
   c.def("__getitem__", [](T &self, const Variable &condition) {
     return extract(self, condition);
   });
-  c.def("__getitem__", [](T &self, const std::tuple<Dim, scipp::index> &index) {
-    return getitem(self, index);
-  });
-  c.def("__getitem__", [](T &self, const std::tuple<Dim, py::slice> &index) {
-    return getitem(self, index);
-  });
-  c.def("__getitem__", [](T &self, const py::ellipsis &index) {
-    return getitem(self, index);
-  });
-  c.def("__setitem__",
-        [](T &self, const scipp::index &index, const py::object &data) {
-          expect_implicit_dimension(self.dims());
-          slicer<T>::template set<std::tuple<Dim, scipp::index>>(
-              self, {self.dim(), index}, data);
-        });
-  c.def("__setitem__",
-        [](T &self, const py::slice &index, const py::object &data) {
-          expect_implicit_dimension(self.dims());
-          expect_positional_index(index);
-          slicer<T>::template set<std::tuple<Dim, py::slice>>(
-              self, {self.dim(), index}, data);
-        });
-  c.def("__setitem__", &slicer<T>::template set<std::tuple<Dim, scipp::index>>);
-  c.def("__setitem__", &slicer<T>::template set<std::tuple<Dim, py::slice>>);
-  c.def("__setitem__", &slicer<T>::template set<py::ellipsis>);
+  // These overloads must also be placed before the ones accepting std::tuple.
+  // Otherwise, pybind11 would call `int()` on the 2nd tuple element
+  // because of `Variable.__int__`.
```

Comment:
What happens now if I do `da[sc.scalar(23)]`? Do we still get a dimension error, as we should?

Reply:
No, this gives
```
scipp._scipp.core.DTypeError: Cannot extract elements based on condition with non-boolean dtype. If you intended to select a range based on a label you must specify the dimension.
```
But that is unchanged, the overload `(T &self, const Variable &condition)` takes precedence over label based slicing. Which, to be honest, I don't even see in this list at all.

Reply:
👍

Reply:
Given the complexity of `__getitem__` arguments, maybe we want to move to manually parsing them like in `Variable.__init__` in order to catch all edge cases and raise proper exceptions?

Reply:
👍 
... but also never change a running system?

Reply:
But proper exception *would* be really useful.